build.zig

```diff
@@ -510,8 +510,8 @@ pub fn build(b: *std.Build) !void {
         .skip_non_native = skip_non_native,
         .skip_libc = skip_libc,
         .use_llvm = use_llvm,
-        // I observed a value of 5136793600 on the M2 CI.
-        .max_rss = 5368709120,
+        // I observed a value of 5605064704 on the M2 CI.
+        .max_rss = 5605064704,
```

Comment:
We add 5% headroom to the last value observed.
```suggestion
        // I observed a value of 5605064704 on the M2 CI.
        .max_rss = 5885317939,
```